SYCL/Config/select_device.cpp

```diff
@@ -253,8 +269,15 @@ int main() {
           std::string ver = plt.get_info<info::platform::version>();
           if (((plt.get_backend() == backend::opencl) &&
                (sycl_be.find("opencl") != std::string::npos)) ||
-              ((plt.get_backend() == backend::ext_oneapi_level_zero) &&
-               (sycl_be.find("level_zero") != std::string::npos))) {
+              (((plt.get_backend() == backend::ext_oneapi_level_zero) ||
+                (plt.get_backend() == backend::level_zero)) &&
```

Comment:
backend::level_zero, backend::hip and backend::cuda values are deprecated. According to SYCL spec we must use backend::ext_oneapi_level_zero, backend::ext_oneapi_hip and backend::ext_oneapi_cuda instead.

This commit did the renaming in the tests some time ago: https://github.com/intel/llvm-test-suite/commit/18c2f11ee2f6a59883ca4b4809af646e8cf39691

So I believe we shouldn't check for deprecated values here.